RedHatInsights/curiosity-frontend
# chore(build): Ephemeral Environments onboarding step 1

## What's included

Files necessary for onboarding curiosity to Ephemeral Environments.

This is necessary for App-SRE Jenkins to build curiosity container. We need curiosity container to deploy curiosity on ephemeral cluster. We need curiosity deployed on cluster to verify if it runs and start working on running iqe tests on EE. So we need this PR in to do anything else related to EEs.

This PR includes #862 and one more commit, both rebased on top of `ci` branch. It's self-contained and should be rebase-able on top of any branch.

### Notes

Current build pipeline uses `verify` script for PR testing and `build:prod` script for production builds. Neither was used, so right now I just made them aliases to existing scripts. But we can change that if we want to have different scripts for App-SRE Jenkins, Travis and local development.

jest cache is disabled based on request of App-SRE (see "PR Check and Build Deploy" on "Frontend Container Migration Onboarding" in internal consoledot docs).

## How to test

```
$ podman run -i --rm -v ./:/workspace:ro,Z -e NODE_BUILD_VERSION=14 -e APP_DIR="" -e IS_PR=true -e CI_ROOT="" quay.io/cloudservices/frontend-build-container:34070c2
<snip>
run-s -l build:pre build:js build:post test:integration
[build:js        ] webpack 5.69.1 compiled with 2 warnings in 43739 ms                 
[build:post      ] $ bash ./scripts/post.sh                                                                                                                                   
[build:post      ] Cleaning up build resources...                                                                                                                             
[test:integration] $ TZ=UTC jest ./tests --no-cache   
[test:integration] PASS tests/html.test.js
[test:integration] PASS tests/code.test.js
[test:integration] PASS tests/dist.test.js
[test:integration] PASS tests/version.test.js
[test:integration] PASS tests/platform.test.js
[test:integration] 
[test:integration] Test Suites: 5 passed, 5 total
[test:integration] Tests:       9 passed, 9 total
[test:integration] Snapshots:   9 passed, 9 total
[test:integration] Time:        3.848 s
[test:integration] Ran all test suites matching /.\/tests/i.
$ echo $?
0
```

Why?

App-SRE Jenkins will use [`frontend-build.sh`@insights-frontend-builder-common](https://github.com/RedHatInsights/insights-frontend-builder-common/blob/master/src/frontend-build.sh) script to build container.

Container is built by `quay.io/cloudservices/frontend-build-container` container, with source code available in `insights-platform/frontend-build-container` repository in internal gitlab instance. That container basically sets specific node version and runs `yarn install --immutable`. Then it will call `yarn verify` (if IS_PR=true) **or** `yarn build:prod` (if IS_PR=false). Various parts afterwards kind of expect `dist` directory to be available, so you can presume `verify` is supposed to run tests and build.

As a side note, we can provide our own build logic in `ci.sh` script.

## Example
N/A

## Updates issue/story
N/A
